File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        l = 0
        r = len(s) - 1
        lst = list(s)
        while(l < r):
            if self.isVowel(s[l]) and self.isVowel(s[r]):
                ch = lst[r]
                lst[r] = lst[l]
                lst[l] = ch
                l += 1
                r -= 1

            elif self.isVowel(lst[l]):
                r -= 1
            elif self.isVowel(lst[r]):
                l += 1
            else:
                l += 1
                r -= 1


        return "".join(lst)

    def isVowel(self, ch) -> bool:
        vowels = "aeiouAEIOU"
        return ch in vowels

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        StringBuilder sb = new StringBuilder(s);
        String vowels = "aeiouAEIOU";
        int l = 0, r = s.length() - 1;
        while(l < r){
            char ch1 = sb.charAt(l);
            char ch2 = sb.charAt(r);
            if(vowels.contains("" + ch1) && vowels.contains("" + ch2)){
                sb.replace(l, l + 1, "" + ch2);
                sb.replace(r, r + 1, "" + ch1);
                l++;
                r--;
            }
            else if(vowels.contains("" + ch1)){
                r--;
            }
            else if(vowels.contains("" + ch2)){
                l++;
            }
            else{
                l++;
                r--;
            }
        }
        return sb.toString();
    }
}